### ISSUE TITLE
ANR immediately after permission request since version 4.0.0 (Android)

### ISSUE THREAD

**Issue by shirozatou** [sebaslogen/resaca#129]
### Example code

```
    <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
```

```
package com.example.myapplication

import android.Manifest
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.animation.AnimatedContent
import androidx.compose.animation.fadeIn
import androidx.compose.animation.fadeOut
import androidx.compose.animation.togetherWith
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.produceState
import androidx.compose.runtime.remember
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

class MainActivity : ComponentActivity() {

    @com.google.accompanist.permissions.ExperimentalPermissionsApi
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent {
            Column(
                modifier = Modifier
                    .fillMaxSize()
                    .padding(60.dp),
                verticalArrangement = Arrangement.spacedBy(20.dp)
            ) {
                val boolState = remember {
                    mutableStateOf(false)
                }

                AnimatedContent(
                    targetState = boolState.value,
                    label = "",
                    transitionSpec = {
                        fadeIn().togetherWith(fadeOut())
                    }
                ) {
                    if (it) {
                        val count = produceState(0) {
                            while (true) {
                                kotlinx.coroutines.delay(1000)
                                value++
                            }
                        }
                        BasicText(
                            text = "ANR??? ${count.value}",
                        )

                        val permissionState =
                            com.google.accompanist.permissions.rememberPermissionState(
                                permission = Manifest.permission.POST_NOTIFICATIONS
                            ) {
                                //noop
                            }
                        LaunchedEffect(Unit) {
                            withContext(Dispatchers.Main) {
                                permissionState.launchPermissionRequest()
                            }
                        }
                    } else {
                        val state = com.sebaslogen.resaca.rememberScoped {
                            mutableStateOf<String?>(null)
                        }
                        BasicText(
                            text = "rememberScoped in using",
                            modifier = Modifier.clickable { state.value = null }
                        )
                    }
                }

                BasicText(
                    text = "permission request dialog",
                    modifier = Modifier
                        .border(1.dp, Color.Black, RoundedCornerShape(50))
                        .clip(RoundedCornerShape(50))
                        .clickable { boolState.value = true }
                        .padding(20.dp)
                )
            }
        }
    }
}

```

### Steps to reproduce

1. Revoke runtime permission or clear App data first
2. Click "permission request dialog"
3. Grant or Deny

### Possible fixes
Try not to use that `rememberScoped`.

**Comment by sebaslogen** [comment]
Thank you very much for reporting it, the bug was indeed introduced in the migration to Compose Multiplatform and I fixed it in [Fix blocking bug on resume from background from resaca 4.0.0](https://github.com/sebaslogen/resaca/commit/e6baf6c6068797ea23a12b367e7e0c7e047ce537)

**Comment by sebaslogen** [comment]
Fix released in [v4.1.1](https://github.com/sebaslogen/resaca/releases/tag/4.1.1)
